ddtrace/contrib/botocore/patch.py

```diff
@@ -453,10 +484,37 @@ def patched_api_call(original_func, instance, args, kwargs):
         span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.botocore.get_analytics_sample_rate())
 
         try:
-            result = original_func(*args, **kwargs)
-            _set_response_metadata_tags(span, result)
+            if endpoint_name == "sqs" and operation == "ReceiveMessage" and config._data_streams_enabled:
+                queue_url = params["QueueUrl"]
+                queue_name = queue_url[queue_url.rfind("/") + 1 :]
+
+                if "MessageAttributeNames" not in args[1]:
+                    args[1].update({"MessageAttributeNames": ["_datadog"]})
+                    args = (args[0], args[1])
+                elif "_datadog" not in args[1]["MessageAttributeNames"]:
+                    args[1].update({"MessageAttributeNames": args[1]["MessageAttributeNames"] + ["_datadog"]})
+                    args = (args[0], args[1])
+
+                result = original_func(*args, **kwargs)
+                _set_response_metadata_tags(span, result)
+
+                for message in result["Messages"]:
+                    try:
+                        pathway = json.loads(message["MessageAttributes"]["_datadog"]["StringValue"])["dd-pathway-ctx"]
```

Comment:
so, in other languages, we add b64 encoded data to the header:
`dd-pathway-ctx-base64`
we can use the same convention here.
Can you check that the encoded data can be decoded in go using the `DecodeStr` function? 
Please, let me know if you need any help for that.
Also, let's put the name `dd-pathway-ctx-base64` as a constant in the core library.
https://github.com/DataDog/data-streams-go/blob/7db3f9c249385c741f5e5e83a01b05a9f4332a81/datastreams/propagator.go#L59